lib/src/utils/node_utils.dart

```diff
@@ -22,3 +24,66 @@ String humanReadableNodeType(AstNode? node) {
 
   return 'Node';
 }
+
+/// Parses the provided type string to extract a [NamedType].
+NamedType parseNamedTypeFromString(String typeString) {
+  try {
+    final namedTypeFinder = _NamedTypeFinder();
+
+    final parseResult = parseString(content: "$typeString _;");
+    parseResult.unit.visitChildren(namedTypeFinder);
+
+    return namedTypeFinder.foundNamedType!;
+  } catch (_) {
+    throw Exception("No NamedType could be parsed from the input "
+        "typeString: '$typeString'. Ensure it's a valid Dart "
+        "type declaration.");
+  }
+}
+
+class _NamedTypeFinder extends GeneralizingAstVisitor<void> {
+  NamedType? _foundNamedType;
+
+  NamedType? get foundNamedType => _foundNamedType;
+
+  @override
+  void visitNamedType(NamedType namedType) {
+    _foundNamedType ??= namedType;
+  }
+}
```

Comment:
Let's place this in a separate file, to limit `_NamedTypeFinder` visibility scope.